docs/develop/python/versioning.mdx

```diff
@@ -231,43 +231,73 @@ class MyWorkflow:
+A Build ID corresponds to a deployment. If you don't already have one, we recommend a hash of the code--such as a Git SHA--combined with a human-readable timestamp.
+To use [Worker Versioning](/workers#worker-versioning), you need to pass a Build ID to your Java Worker and opt in to Worker Versioning.
 
-1. Determine and assign a Build ID to your built Worker code, and opt in to versioning.
-2. Tell the Task Queue your Worker is listening on that Build ID, and whether it's compatible with an existing Build ID.
+### Assign a Build ID to your Worker and opt into versioning
 
-### Assign a Build ID to your Worker
+You should understand assignment rules before completing this step.
+See the [Worker Versioning Pre-release README.md](https://github.com/temporalio/temporal/blob/main/docs/worker-versioning.md) for more information.
 
-Let's say you've chosen `deadbeef` as your Build ID, which might be a short git commit hash (a reasonable choice as Build ID).
-To assign it in your Worker code, assign at least the following Worker Options:
+To enable worker versioning for your worker, assign the build ID--perhaps from an environment variable--and turn it on.
 
 ```python
 # ...
 worker = Worker(
   task_queue="your_task_queue_name",
-  build_id="deadbeef",
+  build_id=build_id,
   use_worker_versioning=True,
   # ... register workflows & activities, etc
 )
 # ...
 ```
 
-That's all you need to do in your Worker code.
-Importantly, if you start this Worker, it won't receive any tasks yet.
-That's because you need to tell the Task Queue about your Worker's Build ID first.
+:::warning
+
+Importantly, when you start this Worker, it won't receive any tasks until you set up assignment rules.
+
+:::
+
+### Specify versions for Activities, Child Workflows, and Continue-as-New Workflows
+
+:::caution
+
+Python support for this feature is under construction!
+
+:::
+
+By default, Activities, Child Workflows, and Continue-as-New Workflows are run on the build of the workflow that created them if they are also configured to run on the same Task Queue.
+When configured to run on a separate task queue, they will default to using the current assignment rules.
+
+If you want to override this behavior, you can specify your intent via the `versioning_intent` argument available on the methods you use to invoke these commands.
+
+For example, if you want an Activity to use the latest assignment rules rather than inheriting from its parent:
+
+```python
+# ...
+await workflow.execute_activity(
+    say_hello,
+    "hi",
+    versioning_intent=VersioningIntent.USE_ASSIGNMENT_RULES,
+    start_to_close_timeout=timedelta(seconds=5),
+)
+# ...
+```
+
+### Tell the Task Queue about your Worker's Build ID (Deprecated)
 
-### Tell the Task Queue about your Worker's Build ID
+:::caution
+
+This section is for a version of Worker Versioning that is deprecated and will go away at some point. Please redirect your attention to [Worker Versioning](/workers#worker-versioning).
```

Comment:
"This section is for a version of Worker Versioning that is deprecated and will go away at some point" could be confusing. I would suggest rephrasing. For example:

> This section covers an earlier implementation of Worker Versioning, which is now deprecated, and which will be removed in the future.